pandas/core/groupby/ops.py

```diff
@@ -290,10 +290,13 @@ def get_result_dtype(self, dtype: DtypeObj) -> DtypeObj:
                 return np.dtype(np.int64)
             elif isinstance(dtype, (BooleanDtype, _IntegerDtype)):
                 return Int64Dtype()
-        elif how in ["mean", "median", "var"] and isinstance(
-            dtype, (BooleanDtype, _IntegerDtype)
-        ):
-            return Float64Dtype()
+        elif how in ["mean", "median", "var"]:
```

Comment:
this entire routine should be in pandas/core/dtypes/common.py or similar. I think we had an issue about this.

Reply:
This was just moved from dtyes/cast.py in #41065